linalg/dual.hpp

```diff
@@ -310,50 +314,48 @@ dual<value_type, gradient_type> tan(dual<value_type, gradient_type> a)
 }
 
 /** @brief implementation of atan for dual numbers */
-template <typename value_type, typename gradient_type>
+template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> atan(dual<value_type, gradient_type> a)
 {
    return {atan(a.value), a.gradient / (value_type{1} + a.value * a.value)};
 }
 
 /** @brief implementation of exponential function for dual numbers */
-template <typename value_type, typename gradient_type>
+template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> exp(dual<value_type, gradient_type> a)
 {
    return {std::exp(a.value), std::exp(a.value) * a.gradient};
 }
 
 /** @brief implementation of the natural logarithm function for dual numbers */
-template <typename value_type, typename gradient_type>
+template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> log(dual<value_type, gradient_type> a)
 {
    return {std::log(a.value), a.gradient / a.value};
 }
 
 /** @brief implementation of `a` (dual) raised to the `b` (dual) power */
-template <typename value_type, typename gradient_type>
+template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> pow(dual<value_type, gradient_type> a,
                                     dual<value_type, gradient_type> b)
 {
-   value_type value = pow(a.value, b.value);
+   value_type value = std::pow(a.value, b.value);
    return {value, value * (a.gradient * (b.value / a.value) + b.gradient * std::log(a.value))};
 }
 
 /** @brief implementation of `a` (non-dual) raised to the `b` (dual) power */
-template <typename value_type, typename gradient_type>
+template <typename value_type, typename gradient_type> MFEM_HOST_DEVICE
 dual<value_type, gradient_type> pow(double a, dual<value_type, gradient_type> b)
 {
-   using std::pow;
-   value_type value = pow(a, b.value);
+   value_type value = std::pow(a, b.value);
```

Comment:
This is not equivalent and might actually introduce errors. Why the switch to `std::pow`?

Reply:
I'm hitting with `nvcc`: 

- `error: no instance of overloaded function "mfem::internal::pow" matches the argument list`
- `argument types are: (double, double)`

I'll revert all `std::` and will keep the only one I needed, commented.

Reply:
maybe it needs a specialized version of pow that uses `std::pow` @samuelpmishLLNL?

Reply:
I added such specialized version - looks better.

Reply:
`done`